pkg/processmanager/v1/process_manager.go

```diff
@@ -0,0 +1,399 @@
+package processmanager
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/goradd/maps"
+	containercollection "github.com/inspektor-gadget/inspektor-gadget/pkg/container-collection"
+	"github.com/prometheus/procfs"
+
+	apitypes "github.com/armosec/armoapi-go/armotypes"
+	tracerexectype "github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/trace/exec/types"
+	"github.com/kubescape/go-logger"
+	"github.com/kubescape/go-logger/helpers"
+	"github.com/kubescape/node-agent/pkg/utils"
+)
+
+const (
+	cleanupInterval = 1 * time.Minute
+	maxTreeDepth    = 50
+)
+
+type ProcessManager struct {
+	containerIdToShimPid maps.SafeMap[string, uint32]
+	processTree          maps.SafeMap[uint32, apitypes.Process]
+	// For testing purposes we allow to override the function that gets process info from /proc.
+	getProcessFromProc func(pid int) (apitypes.Process, error)
+}
+
+func CreateProcessManager(ctx context.Context) *ProcessManager {
+	pm := &ProcessManager{
+		getProcessFromProc: getProcessFromProc,
+	}
+	go pm.startCleanupRoutine(ctx)
+	return pm
+}
+
+// PopulateInitialProcesses scans the /proc filesystem to build the initial process tree
+// for all registered container shim processes. It establishes parent-child relationships
+// between processes and adds them to the process tree if they are descendants of a shim.
+func (p *ProcessManager) PopulateInitialProcesses() error {
+	if len(p.containerIdToShimPid.Keys()) == 0 {
+		return nil
+	}
+
+	fs, err := procfs.NewFS("/proc")
```

Comment:
shouldn't you open `/host/proc` instead?

Reply:
actually I see that you open `/proc` in the rule_manager as well

Reply:
Hmm you might be right.

Reply:
![image](https://github.com/user-attachments/assets/fd813d1b-b271-4608-b76a-a9505233763d)
Nope see this.

Reply:
sababa